### PR TITLE
docs: Module 5 CosmWasm (B9lab updates 4-1)

### PR DESCRIPTION
**Pull request by coldice**
Docs content update for Module 5 (Running my own chain), Section _CosmWasm_ - part of the [b9lab-content](https://github.com/cosmos/sdk-tutorials/tree/master/b9lab-content).

### Content state

**✔️ Completed**

### Technical review

- [x] requested
- [ ] completed

### Language review

- [ ] requested
- [ ] completed

### CODE REVIEW

File: b9lab-content/5-my-own-chain/05-cosmwasm.md [hunk 1]
```diff
@@ -1 +1,232 @@
...
+```bash
+$ wasmd query bank balances $(wasmd keys show wallet --address) --node $RPC
+$ wasmd query bank balances $(wasmd keys show wallet2 --address) --node $RPC
+```
+
+In another location, clone the contract samples:
+
+```bash
+$ git clone https://github.com/CosmWasm/cw-examples
+$ cd cw-examples/contracts/
```
Comment (orkunkl) [inline b9lab-content/5-my-own-chain/05-cosmwasm.md:94]:
```suggestion
$ cd cw-contracts/contracts/
```

File: b9lab-content/5-my-own-chain/05-cosmwasm.md [hunk 2]
```diff
@@ -1 +1,232 @@
...
+$ ls -lh target/wasm32-unknown-unknown/release/cw_nameservice.wasm
+```
+
+As in the case of developing on other blockchains, you want to maintain your smart contract as small as possible. For production purposes, you will optimize the code with a [docker](https://www.docker.com/) container based on an [image provided by CosmWasm](https://hub.docker.com/r/cosmwasm/rust-optimizer/tags):
+
+```bash
+$ docker run --rm -v "$(pwd)":/code \
+  --mount type=volume,source="$(basename "$(pwd)")_cache",target=/code/target \
+  --mount type=volume,source=registry_cache,target=/usr/local/cargo/registry \
+  cosmwasm/rust-optimizer:0.11.5
```
Comment (orkunkl) [inline b9lab-content/5-my-own-chain/05-cosmwasm.md:110]:
```suggestion
  cosmwasm/rust-optimizer:0.12.3
```

**Comment by orkunkl** [comment]
@kaymakf can you dry run the commands in the document?

File: b9lab-content/5-my-own-chain/05-cosmwasm.md [hunk 3]
```diff
@@ -1 +1,232 @@
...
+$ rustup target list --installed
+$ rustup target add wasm32-unknown-unknown
+```
+
+Create a folder and clonde the [wasmd](https://github.com/CosmWasm/wasmd) repository into it:
+
+```bash
+$ git clone https://github.com/CosmWasm/wasmd.git
+$ cd wasmd
+$ git checkout v0.18.0
```
Comment (kaymakf) [inline b9lab-content/5-my-own-chain/05-cosmwasm.md:30]:
```suggestion
$ git checkout v0.18.0
$ make install
```

File: b9lab-content/5-my-own-chain/05-cosmwasm.md [hunk 4]
```diff
@@ -1 +1,232 @@
...
+$ export GENESIS_URL="https://raw.githubusercontent.com/CosmWasm/testnets/master/pebblenet-1/config/genesis.json"
+$ export APP_CONFIG_URL="https://raw.githubusercontent.com/CosmWasm/testnets/master/pebblenet-1/config/app.toml"
+$ export CONFIG_URL="https://raw.githubusercontent.com/CosmWasm/testnets/master/pebblenet-1/config/config.toml"
+
+$ export RPC="https://rpc.pebblenet.cosmwasm.com:443"
+$ export LCD="https://lcd.pebblenet.cosmwasm.com"
+$ export FAUCET="https://faucet.pebblenet.cosmwasm.com"
+
+$ export COSMOVISOR_VERSION="v0.42.9"
+$ export COSMOVISOR_HOME=/root/.wasmd
```
Comment (kaymakf) [inline b9lab-content/5-my-own-chain/05-cosmwasm.md:66]:
```suggestion
$ export COSMOVISOR_HOME=$HOME/.wasmd
```

File: b9lab-content/5-my-own-chain/05-cosmwasm.md (hunk 1 above)
Comment (kaymakf) [reply b9lab-content/5-my-own-chain/05-cosmwasm.md:94]:
```suggestion
$ cd cw-contracts/contracts/nameservice/
```

File: b9lab-content/5-my-own-chain/05-cosmwasm.md [hunk 5]
```diff
@@ -1 +1,232 @@
...
+Compare the result:
+
+```bash
+$ ls -allh artifacts/cw_nameservice.wasm 
+```
+
+Now upload the contract to the blockchain:
+
+```bash
+$ wasmd tx wasm store artifacts/cw_nameservice.wasm --from wallet --node $RPC --chain-id pebblenet-1 --gas-prices 0.001upebble --gas auto --gas-adjustment 1.3
```
Comment (kaymakf) [inline b9lab-content/5-my-own-chain/05-cosmwasm.md:122]:
```suggestion
$ RES=$(wasmd tx wasm store artifacts/cw_nameservice.wasm --from wallet --node $RPC --chain-id pebblenet-1 --gas-prices 0.001upebble --gas auto --gas-adjustment 1.3)
$ CODE_ID=$(echo $RES | jq -r '.logs[0].events[-1].attributes[0].value')
```
might be easier to read

**Review by kaymakf — CHANGES_REQUESTED** [review]

File: b9lab-content/5-my-own-chain/05-cosmwasm.md [hunk 6]
```diff
@@ -1 +1,232 @@
-# CosmWasm
\ No newline at end of file
+# CosmWasm
```
[the remark this answers is not in the document]
Comment (CitMC) [reply b9lab-content/5-my-own-chain/05-cosmwasm.md:1]: I believe this is a very good idea! :) Just to make sure everything is always in its most recent version.

[the remark this answers is not in the document]
Comment (orkunkl) [reply b9lab-content/5-my-own-chain/05-cosmwasm.md:8]: We have switched to `sandynet`. Let's setup the script for making everything easy for you guys.

[the remark this answers is not in the document]
Comment (orkunkl) [reply b9lab-content/5-my-own-chain/05-cosmwasm.md:8]: We don't have a one pager like this rather a directory. https://github.com/InterWasm/docs/tree/main/docs/02-getting-started.
Re: compatibility, I don't think there will be a problem.

**Comment by coldice** [comment]
Merged in b9lab-content-updates-rc1 (no diff to this branch)

**Pull request closed without merging**